### PR TITLE
[M68k] fix call frame destruction elimination

### LINKED ISSUE #106213: [M68k] function epilogue pops stack too far
The generated epilogue for functions that return structs adds 4 to the stack pointer despite there being no corresponding subtraction in the prologue.

A minimal example for this issue would be

```
define internal void @test(ptr sret([10 x i8]) align 2 %_0) unnamed_addr #1 {
start:
  ret void
}
```

which produces the following code (using `llc -O0 -march=m68k`):

```
        .p2align        1                               ; -- Begin function test
        .type   test,@function
test:                                   ; @test
        .cfi_startproc
; %bb.0:                                ; %start
        move.l  (4,%sp), %d0
        move.l  (%sp), %a1
        adda.l  #4, %sp
        move.l  %a1, (%sp)
        rts
.Lfunc_end2:
        .size   test, .Lfunc_end2-test
        .cfi_endproc
                                        ; -- End function
        .section        ".note.GNU-stack","",@progbits
```

The same issue occurs with longer functions that do actually use the stack.

### PR DESCRIPTION
**Pull request by TechnoElf**
Fixes #106213.

This adjusts `eliminateCallFramePseudoInstr` to match the behaviour of the X86 backend.

### CODE REVIEW

File: llvm/lib/Target/M68k/M68kFrameLowering.cpp [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
 using namespace llvm;
 
+#define DEBUG_TYPE "M68k-frame"
```
Comment (mshockwave) [inline llvm/lib/Target/M68k/M68kFrameLowering.cpp:36]: nit: please start with lower case "m68k-frame"

**Review by mshockwave — APPROVED** [review]
LGTM. I would suggest to be a little more explicit in the commit message about the original issue, for instance "fix call frame destruction elimination when returning structs"

File: llvm/lib/Target/M68k/M68kFrameLowering.cpp (hunk 1 above)
Comment (TechnoElf) [reply llvm/lib/Target/M68k/M68kFrameLowering.cpp]: Done

**Comment by glaubitz** [comment]
@mshockwave That commit message looks good now, I guess?

**Comment by mshockwave** [comment]
> @mshockwave That commit message looks good now, I guess?

If using Github's web interface to merge, it actually uses the PR description.

(@TechnoElf FYI)

**Pull request merged**